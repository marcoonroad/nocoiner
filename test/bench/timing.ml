open Core_bench.Bench
module Command = Core.Command

let reveals c o =
  try ignore @@ Nocoiner.reveal ~commitment:c ~opening:o; true
  with Nocoiner.Reasons.BindingFailure -> false

let _RIGHT_SECRET = "P = NP would prove God's existence."
let _WRONG_SECRET = "The Quantum Nature is just Godel..."

let (_RIGHT_C, _RIGHT_O) = Nocoiner.commit _RIGHT_SECRET
let (_WRONG_C, _WRONG_O) = Nocoiner.commit _WRONG_SECRET

let __test_case_01 ( ) = assert (reveals _RIGHT_C _RIGHT_O)
let __test_case_02 ( ) = assert (not (reveals _WRONG_C _RIGHT_O))
let __test_case_03 ( ) = assert (not (reveals _RIGHT_C _WRONG_O))

let _TEST_NAME_01 = "bound opening"
let _TEST_NAME_02 = "unbound commitment"
let _TEST_NAME_03 = "unbound opening"

let __test_01 = Test.create ~name:_TEST_NAME_01 __test_case_01
let __test_02 = Test.create ~name:_TEST_NAME_02 __test_case_02
let __test_03 = Test.create ~name:_TEST_NAME_03 __test_case_03

let suite = [ __test_01; __test_02; __test_03 ]

let _ = Command.run @@ make_command suite

