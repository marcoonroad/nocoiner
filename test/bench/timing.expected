Estimated testing time 45s (3 benchmarks x 15s). Change using '-quota'.
bound opening: Total time taken 15.160114288330078s (155 samples, max runs 155).
unbound commitment: Total time taken 15.044508934020996s (157 samples, max runs 157).
unbound opening: Total time taken 15.120289325714111s (158 samples, max runs 158).
                                                                                                                                                                                               
  Name                 Runs @ Samples   Time/Run   Cycls/Run   mWd/Run   mWd Overhd   mjWd/Run   mjWd Overhd   Prom/Run   Prom Overhd    mGC/Run   mjGC/Run   Comp/Run   Percentage   Speedup  
 -------------------- ---------------- ---------- ----------- --------- ------------ ---------- ------------- ---------- ------------- ---------- ---------- ---------- ------------ --------- 
  bound opening             155 @ 155   817.27us    849.94kc   17.66kw       26.42w     18.06w      -167.38w     18.06w      -167.38w   62.60e-3    0.00e-9    0.00e-9      100.00%      1.05  
  unbound commitment        157 @ 157   780.65us    811.84kc   17.57kw       28.35w     18.89w      -175.07w     18.89w      -175.07w   62.27e-3    0.00e-9    0.00e-9       95.52%      1.00  
  unbound opening           158 @ 158   790.64us    822.23kc   17.57kw       28.42w     18.92w      -176.44w     18.92w      -176.44w   62.30e-3    0.00e-9    0.00e-9       96.74%      1.01  
                                                                                                                                                                                               
